datafusion/functions/src/math/log.rs

```diff
@@ -146,51 +151,74 @@ impl ScalarUDFImpl for LogFunc {
     /// 3. Log(a, a) ===> 1
     fn simplify(
         &self,
-        args: Vec<Expr>,
+        mut args: Vec<Expr>,
         info: &dyn SimplifyInfo,
     ) -> Result<ExprSimplifyResult> {
-        let mut number = &args[0];
-        let mut base =
-            &Expr::Literal(ScalarValue::new_ten(&info.get_data_type(number)?)?);
-        if args.len() == 2 {
-            base = &args[0];
-            number = &args[1];
+        // Args are either
+        // log(number)
+        // log(base, number)
+        let num_args = args.len();
+        if args.len() > 2 {
+            return plan_err!("Expected log to have 1 or 2 arguments, got {num_args}");
```

Comment:
should this be an exec error to match similar patterns elsewhere?

Reply:
lets fix it in follow up. tbh we should unify error messaging for all builtin functions

Reply:
I think the idea here is that simplify gets called as part of planning (not execution) so planning makes sense to me. Though I am not sure what practical difference it makes in this case 🤔